source/common/stats/thread_local_store.cc

```diff
@@ -96,8 +96,8 @@ void ThreadLocalStoreImpl::shutdownThreading() {
 }
 
 void ThreadLocalStoreImpl::mergeHistograms(PostMergeCb merge_complete_cb) {
-  ASSERT(!merge_in_progress_);
   if (!shutting_down_) {
+    ASSERT(!merge_in_progress_);
```

Comment:
i just  moved this check because, I hit the same issue in CI. Since https://github.com/envoyproxy/envoy/pull/3289 fixes it properly we should be good.

Reply:
It'd be great to get that PR in first to avoid affecting this one :)